cellprofiler/modules/exporttocellh5.py

```diff
@@ -306,21 +310,19 @@ def handle_interaction(self, master_file, pid, path):
         master_dict = self.get_dictionary().setdefault(master_file, {})
         if pid not in master_dict:
             md_head, md_tail = os.path.splitext(master_file)
-            subfile = "%s_%s%s" % (md_head, str(pid), md_tail)
```

Comment:
I am worried that this will create thousands of small files in a single directory. My goal here was to have one .h5 file per worker and let the worker write multiple records to that file. Also, the user may not have site metadata and can choose a site metadata item of "None". The code fails in that case because the path has only two items. Would everything work if this line were subfile="_".join([md_head] + path + [md_tail])?